tests/Casts/BooleanTest.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MongoDB\Laravel\Tests\Casts;
+
+use MongoDB\Laravel\Tests\Models\Casting;
+use MongoDB\Laravel\Tests\TestCase;
+
+class BooleanTest extends TestCase
+{
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        Casting::truncate();
+    }
+
+    public function testBool(): void
+    {
+        $model = Casting::query()->create(['booleanValue' => true]);
+
+        self::assertIsBool($model->booleanValue);
+        self::assertEquals(true, $model->booleanValue);
+
+        $model->update(['booleanValue' => false]);
+        $check = Casting::query()->find($model->_id);
+
+        self::assertIsBool($check->booleanValue);
+        self::assertEquals(false, $check->booleanValue);
+
+        $model->update(['booleanValue' => 1]);
+        $check = Casting::query()->find($model->_id);
```

Comment:
You should check the value just after the update, and before refetching it.

Reply:
When we will use refetched data for testing?

Reply:
Adding an assertion, just like this:
```suggestion
        self::assertSame(true, $model->booleanValue);
        $check = Casting::query()->find($model->_id);
```

Reply:
Done.